pkg/kubernetes/deployer.go

```diff
@@ -0,0 +1,55 @@
+package kubernetes
+
+import (
+	"fmt"
+
+	apps "k8s.io/api/apps/v1"
+	core "k8s.io/api/core/v1"
+)
+
+func (k *Kubernetes) GetDeployerResult(namespace string, pod *core.Pod) (string, error, string, error) {
+	fmt.Println("Get Deploy result")
+	addresses, addressError := k.GetFileFromPod(namespace, pod, "/opt/optimism/packages/tokamak/contracts/genesis/addresses.json")
+	stateDump, stateError := k.GetFileFromPod(namespace, pod, "/opt/optimism/packages/tokamak/contracts/genesis/state-dump.latest.json")
+	return addresses, addressError, stateDump, stateError
+}
+
+func (k *Kubernetes) CreateDeployer(namespace string, name string) (*apps.Deployment, error) {
+	fmt.Println("Create deployer: ", name)
+	object, _ := BuildObjectFromYamlFile("./deployments/deployer/deployment.yaml")
+	deployment, success := ConvertToDeployment(object)
+	if !success {
+		panic("Failed to convert to deployment object")
+	}
+	deployment.Name = name
+	deployment.Spec.Selector.MatchLabels = map[string]string{"app": name}
+	deployment.Spec.Template.ObjectMeta.Labels = map[string]string{"app": name}
+
+	var deployerCreationErr error
+	for i := 0; i < 5; i++ {
+		_, deployerCreationErr = k.CreateDeployment(namespace, deployment)
+		if deployerCreationErr == nil {
+			deployerCreationErr = k.WaitingDeploymentCreated(namespace, name)
+			if deployerCreationErr == nil {
+				break
+			}
+		}
+	}
+	return deployment, deployerCreationErr
+}
+
+func (k *Kubernetes) DeleteDeployer(namespace string, name string) error {
+	return k.DeleteDeployment(namespace, name)
+}
+
+func (k *Kubernetes) CreateConfigMapForDeployer(namespace string, rpc string, targetNetwork string, deployKey string) {
```

Comment:
I think we can make a new function to use in common. The name I recommend `CreateConfigmapWithConfig(namespace string, config map[string]string)`. And we can move this function to `kubernetes/controller.go`

Reply:
In fact, I already create a common function in kubernetes/controller.go named OverrideConfigMapFromTemplate. I will use OverrideConfigMapFromTemplate directly

Reply:
Thank you very much, Brave! 🥰

Reply:
I updated!